content/develop/concepts/configuration/theming-colors-and-borders.md

```diff
@@ -0,0 +1,126 @@
+
+- A CSS [`<named-color>`](https://developer.mozilla.org/en-US/docs/Web/CSS/named-color) like `"darkblue"` or `"maroon"`.
+- A HEX string like `"#483d8b"` or `"#6A5ACD"`.
+- An RGB string like `"rgb(106, 90, 205)"` or `"RGB(70, 130, 180)"`.
+- An HSL string like `"hsl(248, 53%, 58%)"` or `"HSL(147, 50%, 47%)"`.
+
+<Tip>
+
+While it is possible for you to specify an alpha value for your colors, this isn't recommended. Streamlit adjusts the alpha value of colors to ensure contextually appropriate shading between background and foreground.
+
+</Tip>
+
+## Default Streamlit colors
+
+Streamlit comes with two preconfigured themes: light and dark. If you don't specify a theme, for each user, Streamlit will attempt to use the preconfigured theme that best matches their browser settings.
+
+## Color and border configuration options
+
+Most theme configuration options can be set for your whole app, with the option to override it with a different value for the sidebar. For example, your app's primary color (`primaryColor`) is used to highlight interactive elements and show focus. If you set `theme.primaryColor`, this will change the primary color for your whole app. However, if you set `theme.sidebar.primaryColor`, this will override `theme.primaryColor` in the sidebar, allowing you to use two different primary colors.
+
+The following two configuration options can only be applied to the whole app:
+
+- `theme.base` sets the default colors for your app's theme to match one of Streamlit's two default themes (`"light"` or `"dark"`).
+- `theme.showSidebarBorder` sets the visibility of the border between the sidebar and the main body of your app.
+
+The following configuration options can be set separately for the sidebar by using the `[theme.sidebar]` table instead of the `[theme]` table in `config.toml`:
+
+- `theme.primaryColor`
+- `theme.backgroundColor`
+- `theme.secondaryBackgroundColor`
+- `theme.textColor`
+- `theme.linkColor`
+- `theme.codeBackgroundColor`
+- `theme.baseRadius`
+- `theme.borderColor`
+- `theme.showWidgetBorder`
+
+For brevity, on the rest of this page, theming configuration options will not include the `theme.` or `theme.sidebar.` prefix.
+
+### `primaryColor`
+
+`primaryColor` defines the accent color most often used throughout your Streamlit
+app. The following features and effects use your primary color:
+
+- Button hover effects.
+- Elements in focus.
+- Selected elements.
+
+<Tip>
+
+When your primary color is used as a background, Streamlit changes the text color to white. For example, this happens for `type="primary"` buttons and for selected items in `st.multiselect`.
+
+For legibility, always choose a primary color that is dark enough to contrast well with white text.
+
+</Tip>
+
+### `backgroundColor`, `secondaryBackgroundColor`, and `codeBackgroundColor`
+
+`backgroundColor` defines the background color of your app. `secondaryBackgroundColor` defines the background color where contrast is needed.
```

Comment:
"where contrast is needed" sounds a bit odd/difficult to understand. Maybe just directly list the places where we use it below? E.g. something like:

```
`secondaryBackgroundColor` defines the background color of various other elements, e.g.:

- ...
```

Or even just:
```
`secondaryBackgroundColor defines the background color of input widgets and various other places like the headers of `st.dataframe` and `st.help` or the background of `st.code`. 
```